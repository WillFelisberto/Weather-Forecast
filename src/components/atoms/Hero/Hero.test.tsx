import { render, screen } from '@/tests/test-utils';

import { Hero } from '.';

describe('Hero Component', () => {
  it('should render the title and highlight correctly', () => {
    const { container } = render(<Hero title="Seeing the weather" highlight="Dark Weather" />);
    expect(screen.getByText(/Seeing the weather/i)).toBeInTheDocument();
    expect(screen.getByText(/Dark Weather/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the subtitle if provided', () => {
    render(
      <Hero
        title="Seeing the weather"
        highlight="Dark Weather"
        subtitle="Stay updated with real-time weather forecasts."
      />
    );
    expect(screen.getByText('Stay updated with real-time weather forecasts.')).toBeInTheDocument();
  });

  it('should not render the subtitle if not provided', () => {
    render(<Hero title="Seeing the weather" highlight="Dark Weather" />);
    expect(
      screen.queryByText('Stay updated with real-time weather forecasts.')
    ).not.toBeInTheDocument();
  });

  it('should apply the custom className', () => {
    render(<Hero title="Seeing the weather" highlight="Dark Weather" className="custom-class" />);
    const container = screen.getByText(/Seeing the weather/i).parentElement;
    expect(container).toHaveClass('custom-class');
  });
});
