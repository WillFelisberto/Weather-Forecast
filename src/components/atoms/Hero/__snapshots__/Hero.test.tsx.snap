// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component should render the title and highlight correctly 1`] = `
<div>
  <div
    class="text-center text-white"
  >
    <h1
      class="text-4xl md:text-5xl font-bold font-secondary leading-[65px]"
    >
      Seeing the weather
       
      <span
        class="bg-gradient-to-r from-[#CAECFF] to-[#78CFFF] bg-clip-text text-transparent"
      >
        Dark Weather
      </span>
      !
    </h1>
  </div>
</div>
`;
